app/helpers/sessions_helper.rb

```diff
@@ -4,8 +4,14 @@ def log_in(user)
   end
 
   def current_user
-    if session[:user_id]
-      @current_user ||= User.find_by(id: session[:user_id])
+    if (user_id = session[:user_id])
```

Comment:
There is soo much happening here

extract this to a different method and ensure that the method does only one task all the time.

`
   user = User.find(user_id)
      if user && user.authenticated?(cookies[:remember_token])
        log_in(user)
        @current_user = user
      end
`